src/python/zquantum/core/wip/circuit/conversions/qiskit_conversions_test.py

```diff
@@ -0,0 +1,227 @@
+import pytest
+import qiskit
+import numpy as np
+import sympy
+from .. import (
+    X,
+    Y,
+    Z,
+    I,
+    T,
+    H,
+    CNOT,
+    CZ,
+    SWAP,
+    ISWAP,
+    RX,
+    RY,
+    RZ,
+    PHASE,
+    CPHASE,
+    XX,
+    YY,
+    ZZ,
+)
+from .qiskit_conversions import convert_to_qiskit, convert_from_qiskit, qiskit_qubit
+
+
+EXAMPLE_SYMBOLIC_ANGLES = [
+    (sympy.Symbol("theta"), qiskit.circuit.Parameter("theta")),
+    (
+        sympy.Symbol("x") + sympy.Symbol("y"),
+        qiskit.circuit.Parameter("x") + qiskit.circuit.Parameter("y"),
+    ),
+    (0.5 * sympy.Symbol("phi") + 1, 0.5 * qiskit.circuit.Parameter("phi") + 1),
+]
+
+
+EQUIVALENT_NONPARAMETRIC_SINGLE_QUBIT_GATES = [
+    (X, qiskit.extensions.XGate),
+    (Y, qiskit.extensions.YGate),
+    (Z, qiskit.extensions.ZGate),
+    (H, qiskit.extensions.HGate),
+    (I, qiskit.extensions.IGate),
+    (T, qiskit.extensions.TGate),
+]
+
+
+EQUIVALENT_NONPARAMETRIC_TWO_QUBIT_GATES = [
+    # TODO: use non-deprecated CXGate
+    (CNOT, qiskit.extensions.CXGate),
+    (CZ, qiskit.extensions.CZGate),
+    (SWAP, qiskit.extensions.SwapGate),
+    (ISWAP, qiskit.extensions.iSwapGate),
+]
+
+
+EQUIVALENT_SINGLE_QUBIT_ROTATION_GATES = [
+    (RX, qiskit.extensions.RXGate),
+    (RY, qiskit.extensions.RYGate),
+    (RZ, qiskit.extensions.RZGate),
+    (PHASE, qiskit.extensions.PhaseGate),
+]
+
+
+EQUIVALENT_TWO_QUBIT_ROTATION_GATES = [
+    (CPHASE, qiskit.extensions.CPhaseGate),
+    (XX, qiskit.extensions.RXXGate),
+    (YY, qiskit.extensions.RYYGate),
+    (ZZ, qiskit.extensions.RZZGate),
+]
+
+
+TWO_QUBIT_SWAP_MATRIX = np.array(
```

Comment:
This is *incidentally* the same matrix as the matrix of the swap gate. But it serves a different purpose. Namely: this is a transition matrix from qiskit matrices to matrices of everything else that has the opposite ordering of qubits. I'd leave it here because it shows that we are not applying some gate, but reinterpreting a matrix in different basis.

Reply:
Ok 👍